File: luci-app-ekooneplstat/files/usr/lib/lua/luci/model/cbi/ekooneplstat.lua
--[[

LuCI eko.one.pl services

]]--
local fs = require "nixio.fs"
require("luci.sys")

local m, s, tmp

local token = luci.sys.exec("/sbin/stat.sh token | tr '\n' ' '")
local thisbutton = "&nbsp;&nbsp;&nbsp;<input type=\"button\" class=\"cbi-button cbi-button-apply\" value=\" " .. translate("This router statistics") .. " \" onclick=\"window.open('http://dl.eko.one.pl/cgi-bin/router.cgi?token=" .. token .. "')\"/>"
local allbutton = "<p>&nbsp;</p><input type=\"button\" class=\"cbi-button cbi-button-apply\" value=\" " .. translate("See all statistics") .. " \" onclick=\"window.open('http://dl.eko.one.pl/stat.html')\"/>"

local last = "<p>&nbsp;</p><p>" .. translate("Last update") .. ": "
local file = "/tmp/stat_time.txt"
if file then
	tmp = fs.readfile(file) or "-"
else
	tmp = "-"
end
last = last .. tmp .. "</p>"

m = Map("system", translate("Statistics"),
	"" .. allbutton .. thisbutton .. last)

s = m:section(TypedSection, "system", translate("Settings"))
s.addremove = false
s.anonymous = true

s1 = s:option(Flag, "stat", translate("Send statistical information"))
s1.rmempty = false

function s1.validate(self, value)
	_G.eko1stat = value
	return value
end

m.on_after_commit = function(self)
	luci.sys.call("uci set system.@system[0].stat=" .. _G.eko1stat .. ";/sbin/stat-cron.sh")
end

return m

File: luci-app-ekooneplstat/files/www/luci-static/resources/view/ekooneplstat.js
'use strict';
'require form';
'require fs';
'require ui';
'require uci';
'require view';

function handleAction(ev, token) {
	if (ev === 'all') {
		window.open('https://dl.eko.one.pl/stat.html');
		return
	}

	if (ev === 'this') {
		window.open('https://dl.eko.one.pl/cgi-bin/router.cgi?token=' + token);
		return;
	}
}

return view.extend({
	load: function() {
		return Promise.all([
			L.resolveDefault(fs.exec('/sbin/stat.sh', [ 'token' ]), null),
			L.resolveDefault(fs.read('/tmp/stat_time.txt'), null)
		]);
	},

	render: function(data) {
		var m, s, o;
		var token = data[0].stdout.replace(/(?:\r\n|\r|\n)/g, '');

		var html = _('Sending information about the router for statistical purposes. More information on the') + ' <a href="https://eko.one.pl/forum/viewtopic.php?id=7708" target="_blank">' + _('eko.one.pl forum') + '</a>.';

		m = new form.Map('system', _('Statistics'), html);

		s = m.section(form.NamedSection, 'global');
		s.render = L.bind(function(view, section_id) {
			return E('div', { 'class': 'cbi-section' }, [
				E('h3', _('Information')),
				E('div', { 'class': 'cbi-value' }, [
					E('label', { 'class': 'cbi-value-title', 'style': 'padding-top:0rem' }, _('Last update')),
					E('div', { 'class': 'cbi-value-field', 'id': 'status', 'style': 'color:#37c' }, data[1] ? data[1] : '-')
				]),
				E('div', { class: 'right' }, [
					E('button', {
					'class': 'btn cbi-button cbi-button-apply',
					'click': ui.createHandlerFn(this, function() {
						return handleAction('all', null);
					})
					}, [ _('See all statistics') ]),
					'\xa0\xa0\xa0',
					E('button', {
					'class': 'btn cbi-button cbi-button-apply',
					'id': 'btn_suspend',
					'click': ui.createHandlerFn(this, function() {
						return handleAction('this', token);
					})
					}, [ _('This router statistics') ])
				])
			]);
		}, o, this);

		s = m.section(form.TypedSection, 'system', _('Settings'));
		s.anonymous = true;
		s.addremove = false;

		o = s.option(form.Flag, 'stat', _('Send statistical information'));
		o.rmempty = false;

		return m.render();
	},

	handleSave: function(ev) {
		var value = (document.querySelector('input[type=checkbox]').checked || '0');
		uci.set('system', '@system[0]', 'stat', value);
		uci.save();
		uci.apply();
		fs.exec('/sbin/stat-cron.sh');
	},
	handleSaveApply: null,
	handleReset: null
});
